### PR TITLE
lp1933001: Preserve phase of decoded FLAC samples

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1933001

Caused by a wrong sign of the scale factor. The denominator is negative and not positive as assumed.

- Negated nominator of the constant factor
- Improved readability of the code

### CODE REVIEW

**Comment by uklotzde** [comment]
Force push: Forgot to push the renaming of the constant.

File: src/sources/soundsourceflac.cpp [hunk 1]
```diff
@@ -403,12 +403,18 @@ namespace {
...
         (static_cast<FLAC__int32>(1) << std::numeric_limits<FLAC__int32>::digits);
 
 inline CSAMPLE convertDecodedSample(FLAC__int32 decodedSample, int bitsPerSample) {
-    DEBUG_ASSERT(std::numeric_limits<FLAC__int32>::is_signed);
-    return (decodedSample << ((std::numeric_limits<FLAC__int32>::digits + 1) - bitsPerSample)) * kSampleScaleFactor;
+    static_assert(std::numeric_limits<FLAC__int32>::is_signed);
+    static_assert(kSampleUnshiftFactor > CSAMPLE_ZERO, "preserve phase of decoded signal");
+    const auto shiftLeft = (std::numeric_limits<FLAC__int32>::digits + 1) - bitsPerSample;
+    const auto shiftedSample = decodedSample << shiftLeft;
+    return shiftedSample * kSampleUnshiftFactor;
```
Comment (daschuer) [inline src/sources/soundsourceflac.cpp:417]: I think this deserves a test. 

Lets assume 8 bitsPerSample where the minimum is  -128 (0x000000FF)
shiftLeft = 31 + 1 - 8 = 24
shiftedSample = decodedSample << shiftLeft
-128 -> 0x0xFF000000
kSampleUnshiftFactor = 1 / 0x8000000
0xFF000000 / 0x8000000 != -1.0f 

Test Failed :-(

```suggestion
    return -CSAMPLE_PEAK * shiftedSample / (static_cast<FLAC__int32>(-1) << shiftLeft); 
```

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/sources/soundsourceflac.cpp (hunk 1 above)
Comment (uklotzde) [reply src/sources/soundsourceflac.cpp]: Indeed, I have used a confusing naming for the constant and temporary variables. Fixed.

Your example is wrong: 0x000000FF = -1, 0x00000080 = -128

**Review by daschuer — APPROVED** [review]
LGTM, thank you.

**Pull request merged**